### ISSUE TITLE
Incorrect VoiceOver behavior when selecting text in editor

### ISSUE THREAD

**Issue by sevapopov2** [microsoft/vscode#162152]
Type: <b>Bug</b>

When selecting text with VoiceOver turned on, it announces all previously selected text instead of announcing the line that has just been selected.
Steps to reproduce:
1. Launch VoiceOver on Mac by pressing Command+f5
2. Launch VSCode.
3. Open a new or existing file in editor.
4. When focus is in the edit field, press shift+down arrow several times to select several lines. You can also press Shift+Option+right/left arrows to select words or Shift+right/left arrows to select by character.
Expected result:
VoiceOver should announce the text that has just been selected after pressing shift+arrows.
Actual result:
VoiceOver announces absolutely all previously selected text started from the first line/text unit.
Impact:
It's impossible for VoiceOver user to select needed portions of code quickly and it slows down the work process a lot.

VS Code version: Code - Insiders 1.72.0-insider (ba515cdbb48470c116884deb6de17981d341ec06, 2022-09-22T07:26:17.254Z)
OS version: Darwin x64 21.6.0
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz (4 x 1800)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|30, 16, 13|
|Memory (System)|8.00GB (0.42GB free)|
|Process Argv|--crash-reporter-id ef3e29e0-3270-44bd-9a23-71f0368383bc|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
python|ms-|2022.14.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vsdfh931cf:30280410
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
bridge0708:30335490
bridge0723:30353136
cmake_vspar411cf:30557515
vsaa593:30376534
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pylantcb52:30568093
vsccc:30564266
pyindex848:30568094

```

</details>

**Comment by isidorn** [comment]
@sevapopov2 thanks a lot for filling this issue. I can reproduce this. This is an upstream issue, either in Chrome or in VoiceOver, I will check with our Chrome/VoiceOver contacts.

**Comment by MarcoZehe** [comment]
For me, this is even worse. It first announces the whole selection before the current chunk was added, as it is being removed. And then repeats the whole selection over, including the new chunk that was added. So the text of the first two lines, if they were selected, is even read twice, before line 3 is also read as now being added to the selection.
@isidorn Did your investigation turn up an existing issue? I have a feeling this has been around for a while, and there should definitely be a Chromium issue on file already. When I helped implement VoiceOver support into Firefox in 2020, we had a similar bug and needed to properly support some specific NSAccessibility interface properties on the platform level to get this right. Since this doesn't affect Windows, it is clearly a MacOS only bug that has its root cause in a faulty implementation in Chromium's Mac support.

**Comment by isidorn** [comment]
Thanks for the ping @MarcoZehe 
Closing as dup of https://github.com/microsoft/vscode/issues/144791
